src/CommitmentsV2/SFA.DAS.CommitmentsV2/Models/Entity.cs

```diff
@@ -0,0 +1,13 @@
+﻿using System;
+using SFA.DAS.UnitOfWork;
+
+namespace SFA.DAS.CommitmentsV2.Models
+{
+    public abstract class Entity
+    {
+        protected void Publish<T>(Func<T> action) where T : class
+        {
+            UnitOfWorkContext.AddEvent(action);
+        }
```

Comment:
I'm unsure how this would work if we subsequently throw an exception after adding the DraftApprenticeship? 

The UnitOfWorkContext adds the event to a static queue, so even if we throw an exception after calling AddDraftApprenticeship(), (which I believe we do), this event will be on the static queue and it would therefore not be deleted when the exception occurs. 

Would it still be processed when another action adds a different event? I can't see how these events are added to the transaction. 

Is the scope being managed by the AsyncLocal type?

Reply:
:+1: